apps/nextjs/src/middlewares/csp.ts

```diff
@@ -154,26 +154,27 @@ const buildCspHeaders = (nonce: string) => {
 
   const cspString = Object.keys(baseCsp)
     .map((policy) => {
-      const value = [...baseCsp[policy]];
+      const value = [...(baseCsp[policy as keyof typeof baseCsp] || [])];
```

Comment:
I suspect all of this is redundant now @codeincontext ?

Reply:
I think something is being conflated here, as the map is per declaration key (like `img-src`), and now `addPolicyIfExists` is doing its own looping for the same thing. The naming of `addPolicyIfExists` suggests that there's something that might not exist, but it isn't taking a key for a declaration name

It doesn't seem to relate to the middleware logs we're trying to solve, so maybe we could address it in a followup PR so we can get the logging fix in today's RC?

Reply:
It turns out that when this is imported from a test then it starts to get TS errors. The approach here does look correct